hardhat.config.ts

```diff
@@ -78,6 +78,7 @@ const userConfig: HardhatUserConfig = {
             allowUnlimitedContractSize: true,
             blockGasLimit: 100000000,
             gas: 100000000,
+            chainId: typeof HARDHAT_CHAIN_ID === "string" ? parseInt(HARDHAT_CHAIN_ID) : 31337,
```

Comment:
nit: Should we support `HARDHAT_CHAIN_ID=` for unsetting variables? I _think_ it would make it have an empty string value. Maybe:
```suggestion
            chainId: typeof HARDHAT_CHAIN_ID === "string" && HARDHAT_CHAIN_ID !== "" ? parseInt(HARDHAT_CHAIN_ID) : 31337,
```